code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -505,17 +506,27 @@
 
 /datum/reagent/medicine/perfluorodecalin
 	name = "Perfluorodecalin"
-	description = "Extremely rapidly restores oxygen deprivation, but inhibits speech. May also heal small amounts of bruising and burns."
+	description = "Restores oxygen deprivation while producing a lesser amount of toxic byproducts. Both scale with exposure to the drug and current amount of oxygen deprivation. Overdose causes toxic byproducts regardless of oxygen deprivation."
 	reagent_state = LIQUID
 	color = "#FF6464"
 	metabolization_rate = 0.25 * REAGENTS_METABOLISM
+	overdose_threshold = 35 // at least 2 full syringes +some, this stuff is nasty if left in for long
 
 /datum/reagent/medicine/perfluorodecalin/on_mob_life(mob/living/carbon/human/M)
-	M.adjustOxyLoss(-12*REM, 0)
-	M.adjustToxLoss(0.3*REM, 0)
+	var/oxycalc = 2.5*REM*current_cycle
+	if(!overdosed)
+		oxycalc = min(oxycalc,M.getOxyLoss()+PERF_BASE_DAMAGE) //if NOT overdosing, we lower our toxdamage to only the damage we actually healed with a minimum of 0.5. IE if we only heal 10 oxygen damage but we COULD have healed 20, we will only take toxdamage for the 10. We would take the toxdamage for the extra 10 if we were overdosing.
+	M.adjustOxyLoss(-oxycalc, 0)
+	M.adjustToxLoss(oxycalc/2.5, 0)
```

Comment:
so every tick it increases the tox damage dealt on each tick by 1? doesnt this mean that 5u is lethal?

Reply:
Sorta ;No (even when OD'd)

It will have the potential to go up 1 per tick. However, it's adjusted for oxygen healed.

Let's say you're at 100 `current_cycle`. That means you can heal 250 Oxydamage. If you only have 50 oxydamage though, you'll only take 20 toxin damage (then cycle 101 you'll prob just take 0.5 since you've healed all oxydamage).

If you're OD'd though you will take 100 toxdamage if you somehow managed to have enough to get to that point.